### ISSUE TITLE
Exception on Files >4GB

### ISSUE THREAD

**Issue by AresScorpio** [pushrax/node-xdcc#5]
Hi due to the fact that the write back uses integer to send back the received bytes it is not possible to retrieve large files. It would be nice if we could together fix this issue. I already tried to shift some bits inside the buffer and send them back but this attemp was less successfull.

**Comment by pushrax** [comment]
I think this is a limitation of the protocol itself – I can't find any documents explaining how to write the length back as anything other than a 4-byte int.

**Comment by AresScorpio** [comment]
Yes, this is a limitation of the protocol. It was not designed to transfer files >4GB since they weren't able to store such large files back then. Never the less it is possible to transmit such large files using MIRC. I will provide a patch for this tomorrow - the patch will be compatible to MIRC, KVIRC and probably some other IRC Clients then you can decide whether you apply it or not.

**Comment by pushrax** [comment]
Great, that works for me.

**Comment by JiPaix** [comment]
https://github.com/pushrax/node-xdcc/blob/442cc7bfa24b55319d0a7fefafc360339f1134e7/lib/xdcc.js#L54-L67

solution:
```js
 var received = 0 
   , sendBuffer = Buffer.alloc(8);
  
 var client = net.connect(details.port, details.ip, function() { 
   self.emit('xdcc-connect', details); 
 }); 
  
 client.on('data', function(data) { 
   file.write(data); 
   received += data.length; 
   sendBuffer.writeBigUint64BE(BigInt(received), 0);
   client.write(sendBuffer); 
   self.emit('xdcc-data', received, details); 
 });
// [...] 
```
